### DISCUSSION TITLE
Personal Access Token生成失败

### DISCUSSION THREAD

**Discussion by WAGFS** [halo-dev/halo#6805] [Q&A] [answered]
我想要通过PAT来通过halo接口认证，从而获取博客文章列表，根据[文档示例](https://github.com/halo-dev/halo/tree/main/docs/authentication#personal-access-token)进行PAT生成时接口报404错误，我是通过docker部署来进行开发环境调试的，不知道这有没有影响。🤔
![image](https://github.com/user-attachments/assets/4f7f02c8-a84e-4651-bd76-2d882d987a33)
目前的解决方案是在console后台创建一个用户，使用这个用户的token进行文章列表获取，但是这种方式响应头中会返回set-cookie且前端无法删除，导致重新进入console后台时管理员账号被顶替成新用户。😞
我的需求是：
1. 能够获取到文章列表
2. 获取到数据时不会被塞入cookie
顺便问一下，记得以前是可以通过`http://localhost:8090/swagger-ui.html`来查看和调试接口，现在貌似无法进入，是地址更换了吗？💔

**Comment by JohnNiang** [comment, marked as answer]
Hi @WAGFS ，这个 API 的 group 部分确实变更了。建议将请求链接更改为：`/apis/uc.api.security.halo.run/v1alpha1/personalaccesstokens` 重试。

如果想要在本地访问 API 文档的话，需要开启 dev profile 才行（添加运行参数：`--spring.profiles.active=dev`）。

**Reply by WAGFS** [reply]
谢谢❤️，这确实有用，我发现它唯一的缺点是在请求头中会暴露授权账号的账号密码（如果知道Basic Token的生成规则的话），不过幸运的是我在翻阅文档的过程中发现可以通过后台个人中心创建PAT，这种会更安全一些，有一说一，之前我没有发现后台里还有这么一个地方😓。
个人中心/创建令牌：https://docs.halo.run/user-guide/user-center/#%E4%B8%AA%E4%BA%BA%E4%BB%A4%E7%89%8C
使用令牌：https://docs.halo.run/developer-guide/restful-api/introduction#basic-auth
有趣的是，调试`PAT`的时候发现`Authorization`传`Bearer`会报错，传`Basic`却能正常响应。
![17284773883616](https://github.com/user-attachments/assets/e837e889-d072-44a1-9398-f83d2c08c775)
![17284774184659](https://github.com/user-attachments/assets/bb189353-dbf1-4dd6-801d-5c235a156a22)

**最后想问一下，想要在本地访问 API 文档的话，使用docker部署有什么环境变量可以支持吗？**

**Comment by guqing** [comment]
> 想要在本地访问 API 文档的话，使用docker部署有什么环境变量可以支持吗？

在本地开发的情况下，可以使用以下 env 参数开启 swagger-ui 同时 `SPRING_THYMELEAF_CACHE` 用于关闭模板引擎缓存便于开发主题时使用，如果不需要可以删除此项，`-v ~/halo-next-dev` 指定了挂载 halo 工作目录到用户目录的 halo-next-dev 目录
```shell
docker run -it -d --name halo -p 8090:8090 -v ~/halo-next-dev/:/root/.halo2/ \                                                                                                                                                     12:20:14
  -e SPRING_THYMELEAF_CACHE=false \
  -e SPRINGDOC_API_DOCS_ENABLED=true \
  -e SPRINGDOC_API_DOCS_VERSION=OPENAPI_3_0 \
  -e SPRINGDOC_SWAGGER_UI_ENABLED=true \
  halohub/halo:2.19
```

**Reply by WAGFS** [reply]
谢谢！它是有用的💕

**Comment by guqing** [comment]
> 调试PAT的时候发现Authorization传Bearer会报错，传Basic却能正常响应。

目前没能复现你说的问题

<img width="694" alt="image" src="https://github.com/user-attachments/assets/8f1c5399-4040-471c-8cf3-4fdeb367dc4b">
<img width="660" alt="image" src="https://github.com/user-attachments/assets/67961d54-bc32-4b3c-9567-3db8263cf426">

**Reply by WAGFS** [reply]
不好意思，是我的问题，我退出登录后再调用接口就正常了。
